Initial makefile and minishift setup

alexeykazakov opened the pull request:
Since it's unclear atm if we still want to go with API Aggregation approach I've extracted the makefile & minishift setup from https://github.com/redhat-developer/git-service/pull/1 to a separate PR (this one).

xcoulon commented on Gopkg.toml:
```diff
@@ -0,0 +1,42 @@
+
+
+[[constraint]]
+  branch = "master"
```
shouldn't we lock the constraint to a specific version here?

alexeykazakov replied:
Good catch. We actually don't need it for now. Cleaned it up in https://github.com/redhat-developer/git-service/pull/2/commits/8469bca48cdf301a39cbc2e86206298cf7674368

xcoulon commented on Gopkg.toml:
```diff
@@ -0,0 +1,42 @@
+  name = "k8s.io/apiserver"
+
+[[constraint]]
+  branch = "master"
```
same question about the version

alexeykazakov replied:
Done in https://github.com/redhat-developer/git-service/pull/2/commits/8469bca48cdf301a39cbc2e86206298cf7674368

xcoulon commented on make/Makefile.dev:
```diff
@@ -0,0 +1,60 @@
+	@oc login -u system:admin
+	@echo "Creating sub resources..."
+	@echo "Creating CRDs..."
+	@oc create -f https://raw.githubusercontent.com/redhat-developer/devopsconsole-operator/master/deploy/crds/devopsconsole_v1alpha1_gitsource_crd.yaml
```
or we could include the `redhat-developer/devopsconsole-operator/deploy/crds` package as a required dependency (using `dep`) and create the CRD from the file in the `vendor` directory. It could bring more stability compared to always applying the latest version from `master` branch on GitHub.

alexeykazakov replied:
I like the idea. But it looks like we can't require a dependency with no go code.
See https://github.com/golang/dep/issues/1306

xcoulon replied:
ah right. As a workaround, we could have a `doc.go` file in that package, maybe?

xcoulon commented on make/Makefile.dev:
```diff
@@ -0,0 +1,60 @@
+
+.PHONY: deploy-apiserver-only
+deploy-apiserver-only:
+	@echo "Creating API Server..."
```
this does not do much, does it?

alexeykazakov replied:
Correct. Just a placeholder for now.

xcoulon replied:
no worries, that's what I thought ;)

xcoulon commented on make/Makefile.dev:
```diff
@@ -0,0 +1,60 @@
+
+.PHONY: create-cr
+create-cr:
+	@echo "Creating Custom Resource..."
```
what else is it supposed to do?

alexeykazakov replied:
It was a placeholder but we probably won't need it anyway. Removed it in https://github.com/redhat-developer/git-service/pull/2/commits/c31c4860ecf89798868488dc28c8def6c9fb4d95

xcoulon commented on make/Makefile.dev:
```diff
@@ -0,0 +1,60 @@
+
+.PHONY: clean-apiserver
+clean-apiserver:
+	@echo "Deleting API Server..."
```
same here

alexeykazakov replied:
This is a placeholder I would like to keep. We will need to deploy and cleanup the server anyway. But I've renamed it to gitserver instead of apiserver in https://github.com/redhat-developer/git-service/pull/2/commits/26b2bf8d54e7712a2986c39b09a55126a21cffa6

xcoulon replied:
ok, fine

xcoulon commented on make/Makefile.dev:
```diff
@@ -0,0 +1,60 @@
+.PHONY: minishift-start
+minishift-start:
+	minishift start --cpus 4 --memory 8GB
+	-eval `minishift docker-env` && oc login -u system:admin
```
you could also add `eval minishift oc-env` to install the `oc` command in the `$PATH`. Also, the syntax may be slightly different on macOS: `eval $(minishift oc-env) && eval $(minishift docker-env)`

xcoulon commented on make/test.mk:
```diff
@@ -0,0 +1,347 @@
+
+# Generate HTML representation (and show in browser) of coverage profile (based on unit tests).
+# Re-runs unit tests if coverage information is outdated.
+.PHONY: coverage-unit-html
```
do we need the HTML output? (I've never used it, personally.)

alexeykazakov replied:
Removed in https://github.com/redhat-developer/git-service/pull/2/commits/f75c256c954915418349b8e6c2bb818403b54a6e

xcoulon replied:
thanks!

xcoulon commented on minishift/README.md:
```diff
@@ -0,0 +1,123 @@
+
+[oc 3.11.0](https://docs.okd.io/latest/cli_reference/get_started_cli.html#installing-the-cli)
+
+[KVM Hypervisor](https://www.linux-kvm.org/page/Downloads)
```
only for Linux

alexeykazakov replied:
Added a comment in https://github.com/redhat-developer/git-service/pull/2/commits/7e27a924e0dd4aa711dbcf9f07e1b7f858a5fa4c

xcoulon replied:
thanks

xcoulon commented on minishift/README.md:
````diff
@@ -0,0 +1,123 @@
+minishift version
+```
+
+#### Install oc
````
not needed thanks to `eval $(minishift oc-env)` for those you use Minishift

alexeykazakov replied:
Changed in https://github.com/redhat-developer/git-service/pull/2/commits/7e27a924e0dd4aa711dbcf9f07e1b7f858a5fa4c

xcoulon replied:
thanks

xcoulon commented on minishift/README.md:
```diff
@@ -0,0 +1,123 @@
+
+After running this, you can verify all containers running container inside minishift using `docker ps`.
+
+**Note**: If you miss this, docker daemon inside minishift couldn't find latest built image which causes `ImagePullBackOff` as we are using `ImagePullPolicy: IfNotPresent`
```
`couldn't find latest built image` -> `will not find the latest image, causing an ImagePullBackOff error ... `

alexeykazakov replied:
Fixed in https://github.com/redhat-developer/git-service/pull/2/commits/7e27a924e0dd4aa711dbcf9f07e1b7f858a5fa4c

xcoulon replied:
thanks

xcoulon commented on pkg/cmd/main.go:
```diff
@@ -0,0 +1,14 @@
+func main() {
+	// Placeholder
+	for {
+		fmt.Println("Busy doing something super important...")
```
definitely! 😂

alexeykazakov commented:
Thanks for the review @xcoulon and @sbryzak! I've addressed the review comments and merging now. If anyone has more comments we can address them in separate PRs.
